### PR TITLE
capstone.sh: Fix shellcheck warning

### PR DESCRIPTION
**Pull request by lzutao**
- Use $( ... ) instead of legacy \` ... \`
- Use git rev-parse instead of grepping and parsing git branch
- Prefer [ p ] && [ q ] as [ p -a q ] is not well defined
- Clone until our needed TIP commit

### CODE REVIEW

File: shlr/capstone.sh [hunk 1]
```diff
@@ -1,58 +1,84 @@
-#!/bin/sh
+#! /bin/sh
```
Comment (XVilka) [inline shlr/capstone.sh]: why space is here?

Comment (lzutao) [reply shlr/capstone.sh]: Space is [allow](https://unix.stackexchange.com/questions/276751/is-space-allowed-between-and-bin-bash-in-shebang) after `#!`. But if you prefer not having space after `#!`, I will change it.

**Comment by radare** [comment]
Never seen that space before anywhere. Kill it

File: shlr/capstone.sh [hunk 2]
```diff
@@ -1,58 +1,84 @@
...
-		HEAD="`git log|head -n1 | awk '{print $2}'`"
-	fi
-	BRANCH="`git branch | grep '*' | cut -d'*' -f2 | cut -d' ' -f2`"
-
-	if [ "${HEAD}" = "${CS_TIP}" ]; then
-		echo "[capstone] Already in TIP, no need to update from git"
-		exit 0
-	fi
+fatal_msg() {
+	>&2 printf '[capstone] %s\n' "$1"
```
Comment (radare) [inline shlr/capstone.sh]: Dafuq?

File: shlr/capstone.sh [hunk 3]
```diff
@@ -1,58 +1,84 @@
...
 }
 
-if [ -d capstone -a ! -d capstone/.git ]; then
-	echo "[capstone] release with no git?"
-	cd capstone
-	for PATCH in ../capstone-patches/* ; do
-		yes n | patch -Rp1 < $PATCH
+patch_capstone() {
+	for patchfile in ../capstone-patches/*.patch ; do
+		yes n | patch --reverse --strip 1 --input "${patchfile}"
```
Comment (radare) [inline shlr/capstone.sh]: That line is full of gnuisms. It will break in all the bsds and macs

Comment (lzutao) [reply shlr/capstone.sh]: :+1:  forgot that, will fix.

File: shlr/capstone.sh (hunk 2 above)
Comment (lzutao) [reply shlr/capstone.sh]: ? See [this](https://stackoverflow.com/a/23550347/5456794). But it's just a matter of style, so I'll change it as you want.

**Pull request merged**